docs/examples/sg_test.rb

```diff
@@ -0,0 +1,100 @@
+with_driver 'aws'
+
+require 'chef/provisioning/aws_driver'
+
+vpc = aws_vpc 'test-vpc' do
+  cidr_block '10.0.0.0/24'
+  internet_gateway true
+end
+
+# Empty
+
+aws_security_group 'test-sg' do
+  vpc 'test-vpc'
+  action :delete
+end
+
+aws_security_group 'test-sg' do
+  vpc 'test-vpc'
+end
+aws_security_group 'test-sg' do
+  vpc 'test-vpc'
+end
+
+# Things we can reference
+aws_security_group 'other-sg' do
+  vpc 'test-vpc'
+end
+
+aws_subnet 'test-subnet' do
+  vpc 'test-vpc'
+end
+
+recipe = self
+ruby_block 'lb' do
```

Comment:
I don't actually know if `load_balancer_options` accepts lazy.  We need to get lazy working through and through.